code/modules/mob/living/life.dm

```diff
@@ -137,24 +137,14 @@
 	return
 
 /mob/living/proc/handle_gravity(delta_time, times_fired)
-	var/gravity = has_gravity()
-	update_gravity(gravity)
-
-	if(gravity > STANDARD_GRAVITY)
-		gravity_animate()
-		handle_high_gravity(gravity, delta_time, times_fired)
-	else if(get_filter("gravity"))
-		remove_filter("gravity")
+	if(gravity_state > STANDARD_GRAVITY)
+		handle_high_gravity(gravity_state, delta_time, times_fired)
 
 /mob/living/proc/gravity_animate()
 	if(!get_filter("gravity"))
 		add_filter("gravity",1,list("type"="motion_blur", "x"=0, "y"=0))
-	INVOKE_ASYNC(src, PROC_REF(gravity_pulse_animation))
-
-/mob/living/proc/gravity_pulse_animation()
-	animate(get_filter("gravity"), y = 1, time = 10)
-	sleep(1 SECONDS)
```

Comment:
Shouldn't we be using sleep here so both animations don't start at the same time?

Reply:
Animations can be queued! If you don't specify an object to animate it will carry on after the last animation finishes. I'm using that + loop=-1 to ensure this animation plays on a loop forever, until the mob stops being under strong gravity